### PR TITLE
refactor(http): format responses as JSON

### PR DESCRIPTION
**Pull request by williamtdr**
Creates a response model to return the result of an endpoint's execution to the client in a JSON-compliant format. **This is a contract-breaking API change.**

The client can now parse the output's success field to see if it can continue with the next action. If not successful, an error object can be returned with an error code and message. This allows for application-specific error codes that don't fit nicely into the HTTP status codes, or provide more information (e.g. failed to start the honey client).

- Adopts a convention of using exports from `http` for status codes, for improved readability.
- Changed the "[...] connecting honeyclient" error to reflect the `err` scenario, namely that the file couldn't be written.
- No longer returns the system error to the client if writing the file fails - this is a bad practice, as it can open security holes, or expose information to the user they shouldn't see. Instead, returns an error message and code, and logs the system error to the console.

Sample output:
```
➜  api git:(refactor/http-response-fmt) ✗ curl localhost:8080/api/honeyclient/create        
{"success":false,"error":{"code":405,"message":"Method \"GET\" is not allowed.\n"}}

➜  api git:(refactor/http-response-fmt) ✗ curl -X POST localhost:8080/api/honeyclient/create
{"success":false,"error":{"code":500,"message":"Failed to write file for honeyclient to consume."}}

➜  api git:(refactor/http-response-fmt) ✗ mkdir ../honeyclient; mkdir ../honeyclient/input  

➜  api git:(refactor/http-response-fmt) ✗ curl -X POST localhost:8080/api/honeyclient/create
{"success":true,"message":"Create ticket '' with ID '1' and URL ''"}   
```

### CODE REVIEW

**Review by rafibarash — COMMENTED** [review]
Great stuff!! Really like your abstraction of the response functionality.

Since the `SetupGetResponse` in the `get_honey_client_by_id` route has the same functionality as the `SetupCreateResponse` in the `create_honey_client` route, and most other routes will need to set multiple headers at one time, I abstracted that functionality as well in commit b8ba4c1. 

I set the headers after the initial error check for the request type. Not sure if that is best practice

File: models/response.go [hunk 1]
```diff
@@ -1,5 +1,10 @@
 package models
 
+type ResponseHeader struct {
```
Comment (williamtdr) [inline models/response.go]: Could we be using the [builtin Header type](https://golang.org/pkg/net/http/#Header) here, exported from net/http?

File: routes/create_honey_client.go [hunk 2]
```diff
@@ -9,29 +9,35 @@ import (
...
-}
-
 func CreateHoneyClient(w http.ResponseWriter, r *http.Request) {
-
-	// Handling requests.
-	SetupCreateResponse(&w, r)
 	// If request type is not current request, return error.
 	if (*r).Method != "POST" {
-		util.WriteHttpErrorCode(w, http.StatusMethodNotAllowed, fmt.Sprintf("Method \"%s\" is not allowed.\n", (*r).Method))
+		msg := fmt.Sprintf("Method \"%s\" is not allowed.", (*r).Method)
```
Comment (williamtdr) [inline routes/create_honey_client.go:15]: Much better, nice :)

File: models/response.go (hunk 1 above)
Comment (williamtdr) [reply models/response.go]: `myHeader := http.Header{"Content-Type": {"Foo"}, "Requires-Authentication": {"bar", "buzz"}}`

Comment (rafibarash) [reply models/response.go]: nice catch, refactored the code to use http.Header instead of custom ResponseHeader in commit 10525b5

**Review by rafibarash — APPROVED** [review]
looks good and working as intended

**Pull request merged**